Node.js 20 workflow causes issues with modbus and mbus binding (related to serialport)

We need to investigate the reason why Node.js 20 is not working as expected

----
A proposal could be to run Node.js 16 and 18 only (for now) and we investigate why mbus and modbus are failing with Node.js 20.
What do you think?

_Originally posted by @danielpeintner in https://github.com/eclipse-thingweb/node-wot/issues/995#issuecomment-1553062342_

For what it's worth, I also get an error with Node.js v18.16.1:

```
npm ERR! npm ERR! code 1
npm ERR! npm ERR! path /home/victor.charpenay/.npm/_cacache/tmp/git-cloneqLIH1l/node_modules/node-mbus
npm ERR! npm ERR! command failed
npm ERR! npm ERR! command sh -c node-gyp rebuild
```

I updated dependencies in my project from `0.7.0` to `eclipse-thingweb/node-wot#0f6c5f8` (to test the fix pushed in this commit) but with`npm i`, I then get the error above.

--- a/.github/workflows/ci-full.yaml
+++ b/.github/workflows/ci-full.yaml
@@ -8,7 +8,7 @@ jobs:
         strategy:
             matrix:
                 os: [macos-latest, ubuntu-latest, windows-latest]
-                node-version: [16.x, 18.x]
+                node-version: [16.x, 18.x, 20.x]
 
         steps:
             - name: Checkout

--- a/.github/workflows/ci-full.yaml
+++ b/.github/workflows/ci-full.yaml
@@ -19,9 +19,6 @@ jobs:
               with:
                   node-version: ${{ matrix.node-version }}
                   cache: "npm"
-            - name: npm 7
-              # npm workspaces requires npm v7 or higher
-              run: npm i -g npm@7 --registry=https://registry.npmjs.org
             - name: Install
               run: npm ci
 
@@ -41,14 +38,10 @@ jobs:
             - name: Checkout
               uses: actions/checkout@v2
 
-            - name: Use Node.js 14
+            - name: Use Node.js 16
               uses: actions/setup-node@v1
               with:
-                  node-version: 14
-
-            - name: npm 7
-              # npm workspaces requires npm v7 or higher
-              run: npm i -g npm@7 --registry=https://registry.npmjs.org
+                  node-version: 16
 
             - name: Install
               run: npm ci
@@ -61,10 +54,10 @@ jobs:
         runs-on: ubuntu-latest
         steps:
             - uses: actions/checkout@v1
-            - name: install node v14
+            - name: install node v16
               uses: actions/setup-node@v1
               with:
-                  node-version: 14
+                  node-version: 16
             - name: verify packages version consistency accross sub-modules
               run: npm run check:versions
 

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -8,7 +8,7 @@ jobs:
         strategy:
             matrix:
                 os: [ubuntu-latest]
-                node-version: [16.x, 18.x]
+                node-version: [16.x, 18.x, 20.x]
 
         steps:
             - name: Checkout

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -19,9 +19,6 @@ jobs:
               with:
                   node-version: ${{ matrix.node-version }}
                   cache: "npm"
-            - name: npm 7
-              # npm workspaces requires npm v7 or higher
-              run: npm i -g npm@7 --registry=https://registry.npmjs.org
             - name: Install
               run: npm ci
 
@@ -41,14 +38,10 @@ jobs:
             - name: Checkout
               uses: actions/checkout@v2
 
-            - name: Use Node.js 14
+            - name: Use Node.js 16
               uses: actions/setup-node@v1
               with:
-                  node-version: 14
-
-            - name: npm 7
-              # npm workspaces requires npm v7 or higher
-              run: npm i -g npm@7 --registry=https://registry.npmjs.org
+                  node-version: 16
 
             - name: Install
               run: npm ci
@@ -61,10 +54,10 @@ jobs:
         runs-on: ubuntu-latest
         steps:
             - uses: actions/checkout@v1
-            - name: install node v14
+            - name: install node v16
               uses: actions/setup-node@v1
               with:
-                  node-version: 14
+                  node-version: 16
             - name: verify packages version consistency accross sub-modules
               run: npm run check:versions
 

--- a/README.md
+++ b/README.md
@@ -100,7 +100,6 @@ cs.addCodec(new MyCodec("application/myType"));
 ### To use with Node.js
 
 > **Warning**: We no longer actively support Node.js version 14 and lower.
-> Node.js version 20 is not yet supported (https://github.com/eclipse-thingweb/node-wot/issues/1004)
 
 -   [Node.js](https://nodejs.org/) version 14+
 -   [npm](https://www.npmjs.com/) version 7+

--- a/packages/binding-mbus/package.json
+++ b/packages/binding-mbus/package.json
@@ -39,7 +39,7 @@
     "dependencies": {
         "@node-wot/core": "0.8.8",
         "@node-wot/td-tools": "0.8.8",
-        "node-mbus": "^1.2.2",
+        "node-mbus": "^2.1.0",
         "wot-typescript-definitions": "0.8.0-SNAPSHOT.26"
     },
     "scripts": {

